### PR TITLE
[SC-89752] create GH actions for ansible-docker-core

### PR DESCRIPTION
**Pull request by vignesh-v3**
currently we use sjoshi/ansible-core image in almost all of our role-* repositories, this PR would start to push images in the GHCR

we now have the images pushed to [GHCR](https://github.com/quiknode-labs/docker-ansible-core/pkgs/container/docker-ansible-core/versions)

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -1,42 +1,129 @@
...
+  build_push_python38_and_later:
+    needs: build_push_before_python38
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      packages: write
+    strategy:
+      matrix:
+        os: [alpine, ubuntu]
+        version: [v2.12, v2.13,v2.15]
```
Comment (McSim85) [inline .github/workflows/docker-image.yml:83]:
```suggestion
        version: [v2.13, v2.15, v2.16]
```

**Review by McSim85 — CHANGES_REQUESTED** [review]
This is so awesome, that you started to work on this @vignesh-v3 
Overall the recommendation - configure dependabot properly, so it will automatically submit new versions.

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -1,42 +1,129 @@
...
-#   push:
-#     branches:
-#       - 'master'
+  schedule:
+    - cron: '0 1 * * 0'
+
+env:
+  IMAGE_NAME: ghcr.io/quiknode-labs/docker-ansible-core
+  LATEST_OS: alpine
+  LATEST_VERSION: v2.13
```
[the remark this answers is not in the document]
Comment (vignesh-v3) [reply .github/workflows/docker-image.yml:14]: its the alpine version

[the remark this answers is not in the document]
Comment (vignesh-v3) [reply .github/workflows/docker-image.yml:14]: @McSim85 the current one which we use here is [v2.13](https://github.com/sjoshi10/docker-ansible-core/blob/main/.github/workflows/ci.yml#L15)

[the remark this answers is not in the document]
Comment (vignesh-v3) [reply .github/workflows/docker-image.yml:14]: updated the ansible to 2.16

**Review by sjoshi10 — APPROVED** [review]
👍

**Comment by vignesh-v3** [comment]
> This is so awesome, that you started to work on this @vignesh-v3 Overall the recommendation - configure dependabot properly, so it will automatically submit new versions.

@McSim85 i pushed a minor pip related change to dependabot.yml, please let me know if we're good.

cc: @sjoshi10

**Review by McSim85 — APPROVED** [review]
Guys, I am approving this PR, because the goal of the PR - moving from Saurab account to quicknode.
But, I think, we should revisit this role later to make
- dependabot handle all version upgrades
- remove unused OS like almalinux8, alpine
- RUN instructions can be also revisited and improve multiple apt commands

**Comment by vignesh-v3** [comment]
going to merge this, we can open a separate story for the above items.
cc: @sjoshi10

**Pull request merged**